### DISCUSSION TITLE
Concatenate in Statamic

### DISCUSSION THREAD

**Discussion by niloc7** [statamic/cms#4830] [Q&A] [answered]
Quick one, how do you Concatenate in Statamic?

```
<title>{{ title ?? site:name }} {{ product_id ensure_left=" | " ?? parent:product_id  ensure_left=" | " ?? "" }} | Amazon</title>
```

How do you join a page's metadata (product_id) together with custom strings "example words to add in"?

Cause ensure_left doesn't work.. and adding periods like PHP doesn't work too..

**Comment by edalzell** [comment]
You can't combine the modifiers w/ the `??` etc like that.

Break it up w/ `{{ if ...}` and/or maybe use a partial here?

**Comment by edalzell** [comment, marked as answer]
Ya I'd use a partial here: `{{ partial:concat id="{ product_id ?? parent:id }" }}` then in there use `ensure_left`
